### ISSUE TITLE
unable update or re-install widevine for netflix

### ISSUE THREAD

**Issue by brunoduc** [emilsvennesson/script.module.inputstreamhelper#146]
install from [Odroid project](https://forum.odroid.com/viewtopic.php?f=98&t=34149)
Debug log :
```
kodi@omega:~# cat /home/kodi/.kodi/temp/kodi.log 
2019-09-02 09:46:53.798 T:2968072736  NOTICE: -----------------------------------------------------------------------
2019-09-02 09:46:53.798 T:2968072736  NOTICE: Starting Kodi (18.3 Git:20190619-89472b7d69-dirty). Platform: Linux ARM 32-bit
2019-09-02 09:46:53.798 T:2968072736  NOTICE: Using Release Kodi x32 build
2019-09-02 09:46:53.798 T:2968072736  NOTICE: Kodi compiled 2019-06-22 by GCC 7.4.0 for Linux ARM 32-bit version 4.15.18 (266002)
2019-09-02 09:46:53.799 T:2968072736  NOTICE: Running on Ubuntu 18.04.3 LTS, kernel: Linux ARM 32-bit version 4.14.133-166
2019-09-02 09:46:53.799 T:2968072736  NOTICE: FFmpeg version/source: 4.0.3-Kodi
2019-09-02 09:46:53.799 T:2968072736  NOTICE: Host CPU: ARMv7 Processor rev 3 (v7l), 8 cores available
2019-09-02 09:46:53.799 T:2968072736  NOTICE: ARM Features: Neon enabled
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://xbmc/ is mapped to: /usr/share/kodi
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://xbmcbin/ is mapped to: /usr/lib/arm-linux-gnueabihf/kodi
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://xbmcbinaddons/ is mapped to: /usr/lib/arm-linux-gnueabihf/kodi/addons
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://masterprofile/ is mapped to: /home/kodi/.kodi/userdata
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://envhome/ is mapped to: /home/kodi
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://home/ is mapped to: /home/kodi/.kodi
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://temp/ is mapped to: /home/kodi/.kodi/temp
2019-09-02 09:46:53.799 T:2968072736  NOTICE: special://logpath/ is mapped to: /home/kodi/.kodi/temp
2019-09-02 09:46:53.799 T:2968072736  NOTICE: The executable running is: /usr/lib/arm-linux-gnueabihf/kodi/kodi.bin
...
2019-09-02 09:46:53.872 T:2968072736  NOTICE: Running database version Addons27
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: audioencoder.kodi.builtin.aac v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: audioencoder.kodi.builtin.wma v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: game.controller.default v1.0.8 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: game.controller.snes v1.0.8 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: inputstream.adaptive v2.3.22 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.global.audioengine v1.0.1 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.global.filesystem v1.0.2 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.global.general v1.0.3 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.global.gui v5.12.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.global.main v1.0.14 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.global.network v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.audiodecoder v2.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.audioencoder v2.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.game v1.1.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.imagedecoder v2.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.inputstream v2.0.8 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.peripheral v1.3.7 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.pvr v5.10.3 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.screensaver v2.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.vfs v2.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.videocodec v1.0.1 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.binary.instance.visualization v2.0.1 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: kodi.resource v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.album.universal v3.1.2 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.artists.universal v4.3.2 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.common.allmusic.com v3.2.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.common.fanart.tv v3.6.1 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.common.imdb.com v3.1.4 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.common.musicbrainz.org v2.2.1 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.common.theaudiodb.com v2.0.2 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.common.themoviedb.org v3.1.9 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.local v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.themoviedb.org v5.1.8 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: metadata.tvshows.themoviedb.org v3.1.3 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: plugin.audio.jambmc v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: plugin.video.netflix v0.15.2 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: repository.castagnait v1.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: repository.kodinerds v6.0.0 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: repository.netflix v1.0.1 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: repository.sandmann79.plugins v1.0.2 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: repository.xbmc.org v3.1.5 installed
2019-09-02 09:46:53.904 T:2968072736  NOTICE: ADDON: resource.images.weathericons.default v1.1.8 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: resource.language.en_gb v2.0.1 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: resource.language.fr_fr v9.0.22 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: resource.uisounds.kodi v1.0.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: screensaver.xbmc.builtin.black v1.0.32 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: screensaver.xbmc.builtin.dim v1.0.55 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.addon.signals v0.0.3 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.certifi v2019.3.9 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.chardet v3.0.4 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.future v0.17.1 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.idna v2.8 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.inputstreamhelper v0.4.1 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.kodi-six v0.1.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.pil v1.1.7 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.pycryptodome v3.4.3 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.pytz v2014.2 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.requests v2.22.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.urllib3 v1.25.3 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.module.xbmcswift2 v2.4.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.openweathermap.maps v0.2.3 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: script.playalbum v3.0.4 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: service.upnext v1.0.1 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: service.xbmc.versioncheck v0.4.6 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: skin.confluence v4.6.5 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: skin.estouchy v2.0.24 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: skin.estuary v2.0.23 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: visualization.shadertoy v1.1.8 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: visualization.spectrum v2.0.3 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: visualization.waveform v2.0.1 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: weather.gismeteo v0.3.3 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: weather.metoffice v2.0.1 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: weather.openweathermap.extended v5.0.2 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: weather.yahoo v5.0.5 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: webinterface.default v18.x-2.4.6 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.addon v18.3.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.core v0.1.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.gui v5.14.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.json v10.3.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.metadata v2.1.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.python v2.26.0 installed
2019-09-02 09:46:53.905 T:2968072736  NOTICE: ADDON: xbmc.webinterface v1.0.0 installed
...
2019-09-02 09:51:13.448 T:2941244256   ERROR: [plugin.video.netflix (1)] Traceback (most recent call last):
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/addon.py", line 109, in <module>
                                                route(filter(None, g.PATH.split('/')))
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/addon.py", line 50, in lazy_login_wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/addon.py", line 70, in route
                                                player.play(pathitems=pathitems[1:])
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/common/videoid.py", line 287, in wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/common/misc_utils.py", line 246, in timing_wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/navigation/player.py", line 57, in play
                                                list_item = get_inputstream_listitem(videoid)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/navigation/player.py", line 102, in get_inputstream_listitem
                                                if not is_helper.check_inputstream():
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 1004, in check_inputstream
                                                return self._check_drm()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 962, in _check_drm
                                                return self.install_widevine()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 745, in install_widevine
                                                return self._install_widevine_arm()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 710, in _install_widevine_arm
                                                self._losetup(bin_path),
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 371, in _losetup
                                                output = self._run_cmd(cmd, sudo=True)
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 324, in _run_cmd
                                                output = subprocess.check_output(cmd, shell=shell, stderr=subprocess.STDOUT)
                                              File "/usr/lib/python2.7/subprocess.py", line 216, in check_output
                                                process = Popen(stdout=PIPE, *popenargs, **kwargs)
                                              File "/usr/lib/python2.7/subprocess.py", line 394, in __init__
                                                errread, errwrite)
                                              File "/usr/lib/python2.7/subprocess.py", line 1047, in _execute_child
                                                raise child_exception
                                            TypeError: execv() arg 2 must contain only strings
kodi@omega:~# 
```

**Comment by brunoduc** [comment]
After create new user, new Netflix install procedure work. Just update widevine problem ...

**Comment by mediaminister** [comment]
@brunoduc Can you test if it works with the previous 0.3.5 version: https://mirrors.kodi.tv/addons/leia/script.module.inputstreamhelper/script.module.inputstreamhelper-0.3.5.zip
To test a previous version, you should first remove the 0.4.1 version and you may have to disable` Install updates automatically`.

**Comment by brunoduc** [comment]
it's work with the previous version after erase folder `.kodi/userdata/addon_data/script.module.inputstreamhelper/`

```
2019-09-02 17:23:08.223 T:2967691808  NOTICE: -----------------------------------------------------------------------
2019-09-02 17:23:08.223 T:2967691808  NOTICE: Starting Kodi (18.3 Git:20190619-89472b7d69-dirty). Platform: Linux ARM 32-bit
2019-09-02 17:23:08.223 T:2967691808  NOTICE: Using Release Kodi x32 build
2019-09-02 17:23:08.223 T:2967691808  NOTICE: Kodi compiled 2019-06-22 by GCC 7.4.0 for Linux ARM 32-bit version 4.15.18 (266002)
2019-09-02 17:23:08.224 T:2967691808  NOTICE: Running on Ubuntu 18.04.3 LTS, kernel: Linux ARM 32-bit version 4.14.133-166
2019-09-02 17:23:08.224 T:2967691808  NOTICE: FFmpeg version/source: 4.0.3-Kodi
2019-09-02 17:23:08.224 T:2967691808  NOTICE: Host CPU: ARMv7 Processor rev 3 (v7l), 8 cores available
2019-09-02 17:23:08.224 T:2967691808  NOTICE: ARM Features: Neon enabled
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://xbmc/ is mapped to: /usr/share/kodi
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://xbmcbin/ is mapped to: /usr/lib/arm-linux-gnueabihf/kodi
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://xbmcbinaddons/ is mapped to: /usr/lib/arm-linux-gnueabihf/kodi/addons
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://masterprofile/ is mapped to: /home/kodi/.kodi/userdata
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://envhome/ is mapped to: /home/kodi
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://home/ is mapped to: /home/kodi/.kodi
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://temp/ is mapped to: /home/kodi/.kodi/temp
2019-09-02 17:23:08.224 T:2967691808  NOTICE: special://logpath/ is mapped to: /home/kodi/.kodi/temp
2019-09-02 17:23:08.224 T:2967691808  NOTICE: The executable running is: /usr/lib/arm-linux-gnueabihf/kodi/kodi.bin
2019-09-02 17:23:08.224 T:2967691808  NOTICE: Local hostname: omega
2019-09-02 17:23:08.224 T:2967691808  NOTICE: Log File is located: /home/kodi/.kodi/temp/kodi.log
...
2019-09-02 17:23:08.334 T:2967691808  NOTICE: ADDON: script.module.inputstreamhelper v0.3.5 installed
```

I try update to 0.4.1

**Comment by brunoduc** [comment]
After update, it want re-update widewine and fail with the same messages
```
...
2019-09-02 17:49:17.806 T:2527527776  NOTICE: ADDON: script.module.inputstreamhelper v0.4.1 installed
...
2019-09-02 17:51:21.115 T:2255479648   ERROR: [plugin.video.netflix (1)] Traceback (most recent call last):
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/addon.py", line 109, in <module>
                                                route(filter(None, g.PATH.split('/')))
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/addon.py", line 50, in lazy_login_wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/addon.py", line 70, in route
                                                player.play(pathitems=pathitems[1:])
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/common/videoid.py", line 287, in wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/common/misc_utils.py", line 246, in timing_wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/navigation/player.py", line 57, in play
                                                list_item = get_inputstream_listitem(videoid)
                                              File "/home/kodi/.kodi/addons/plugin.video.netflix/resources/lib/navigation/player.py", line 102, in get_inputstream_listitem
                                                if not is_helper.check_inputstream():
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 1004, in check_inputstream
                                                return self._check_drm()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 959, in _check_drm
                                                return self._check_widevine()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 882, in _check_widevine
                                                return self.install_widevine()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 745, in install_widevine
                                                return self._install_widevine_arm()
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 710, in _install_widevine_arm
                                                self._losetup(bin_path),
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 371, in _losetup
                                                output = self._run_cmd(cmd, sudo=True)
                                              File "/home/kodi/.kodi/addons/script.module.inputstreamhelper/lib/inputstreamhelper.py", line 324, in _run_cmd
                                                output = subprocess.check_output(cmd, shell=shell, stderr=subprocess.STDOUT)
                                              File "/usr/lib/python2.7/subprocess.py", line 216, in check_output
                                                process = Popen(stdout=PIPE, *popenargs, **kwargs)
                                              File "/usr/lib/python2.7/subprocess.py", line 394, in __init__
                                                errread, errwrite)
                                              File "/usr/lib/python2.7/subprocess.py", line 1047, in _execute_child
                                                raise child_exception
                                            TypeError: execv() arg 2 must contain only strings
```

**Comment by berturion** [comment]
Same issue here on OSMC on a Rpi3. Cant't use Netflix anymore. The process crashes when extracting the downloaded ChromeOS Image and I have to reboot.

**Comment by mediaminister** [comment]
I have a possible fix for this. Can you please test this version with the **Install from zip file** function in Kodi
[script.module.inputstreamhelper-0.4.1-armfix-5dc3e7b.zip](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3566874/script.module.inputstreamhelper-0.4.1-armfix-5dc3e7b.zip)

**Comment by brunoduc** [comment]
sorry doesn't work

>Same issue here on OSMC on a Rpi3.

For me on a Odoid Xu4 work
on a existing user -> remove version 0.4.1
disable addons inputstreamhelper in kodi
quit kodi
`rm -R .kodi/addons/script.module.inputstreamhelper`
`rm -R .kodi/userdata/addon_data/script.module.inputstreamhelper`
install old version [script.module.inputstreamhelper-0.4.1-armfix-5dc3e7b.zip](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3566874/script.module.inputstreamhelper-0.4.1-armfix-5dc3e7b.zip)

on a new user and a new installation
work with official version 0.4.1

i post a full debug log

**Comment by mediaminister** [comment]
Does https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3566874/script.module.inputstreamhelper-0.4.1-armfix-5dc3e7b.zip work or not?
Please confirm this with details about your Kodi system en version.

**Comment by brunoduc** [comment]
full debug log
[kodi.log.txt](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3566921/kodi.log.txt)

**Comment by brunoduc** [comment]
No It's don't work
[kodi.log.txt](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3566930/kodi.log.txt)

**Comment by tito** [comment]
Nop, same error with script.module.inputstreamhelper-0.4.1-armfix-5dc3e7b.zip. Please note i'm on OSMC (aarch64).

Before updating, i was in the process to debug it. I added debug in `_run_cmd`, and the losetup is wrong, the offset passed is `False`. Currently debugging the `_chrome_offset` function.

**Comment by mediaminister** [comment]
Thanks a lot for debugging this. If you could make a pull request with the fix that would be wonderful.

**Comment by tito** [comment]
I'll, just got another issue with netflix, trying to see if it's with new widevine or not.

My current patch:
```
--- script.module.inputstreamhelper/lib/inputstreamhelper.py.orig	2019-09-02 21:29:50.343718358 +0200
+++ script.module.inputstreamhelper/lib/inputstreamhelper.py	2019-09-02 21:30:13.604078578 +0200
@@ -262,7 +262,7 @@
         if self._cmd_exists('parted'):
             cmd = ['parted', '-s', bin_path, 'unit s print']
         else:
-            cmd = ['fdisk', '-l', bin_path]
+            cmd = ['fdisk', '-l', bin_path, '--bytes']
 
         header = None
         best_size = 0
@@ -275,7 +275,7 @@
         # Routine that works for both fdisk and parted
         for line in output['output'].splitlines():
             # Split by whitespace is not sufficient, some header titles use whitespace
-            partition_data = list([_f for _f in re.split(r'\s{2,}', line) if _f])
+            partition_data = list([_f for _f in re.split(r'\s{1,}', line) if _f])
             if not partition_data:
                 continue
 
@@ -298,7 +298,13 @@
 
             # Find the largest partition, and store the offset
             # NOTE: For parted we need to remove the 's' unit
-            partition_size = int((partition.get('Size') or partition.get('Blocks')).replace('s', ''))
+            try:
+                partition_size = int((partition.get('Size') or partition.get('Blocks')).replace('s', ''))
+            except ValueError:
+                # maybe it's another warning message at the end of fdisk, like:
+                # Partition tables entries are not in disk order
+                continue
+
             if partition.get('Name') == 'ROOT-A' or partition_size > best_size:
                 # The name or id is always stored in the first column
                 partition_name = partition_data[0]

```

**Comment by tito** [comment]
I don't have parted, so maybe i broke this part with the splitting to one whitespace.

**Comment by mediaminister** [comment]
I'll test you patch.

**Comment by tito** [comment]
Here is the PR :)

**Comment by mediaminister** [comment]
The PR seems to work okay.
@berturion @Pantastisch @brunoduc Can you please test this version with the PR from @tito:
[script.module.inputstreamhelper-0.4.1-parsing-78a4a1d.zip](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3567014/script.module.inputstreamhelper-0.4.1-parsing-78a4a1d.zip)

**Comment by Turkehlton** [comment]
Hey guys. Same issue here, rpi 3b+, osmc kodi 18.3.
The latest version @mediaminister (with the PR from @tito ) seems to work for me.
You guys are the best, thanks.

**Comment by dagwieers** [comment]
Try release v0.4.0, this was supposed to be released:
https://github.com/emilsvennesson/script.module.inputstreamhelper/releases/download/v0.4.0/script.module.inputstreamhelper-0.4.0.zip
I am confident it works as expected too.

**Comment by brunoduc** [comment]
Le lundi 2 septembre 2019, 21:54:08 CEST mediaminister a écrit :
> The PR seems to work okay.
> @berturion @Pantastisch @brunoduc Can you please test this version with the
> PR from @tito:
> [script.module.inputstreamhelper-0.4.1-parsing-78a4a1d.zip](https://github.
> com/emilsvennesson/script.module.inputstreamhelper/files/3567014/script.modu
> le.inputstreamhelper-0.4.1-parsing-78a4a1d.zip)

thank you,

script.module.inputstreamhelper-0.4.1-parsing-78a4a1d.zip work for me.

**Comment by Arathonk** [comment]
> The PR seems to work okay.
> @berturion @Pantastisch @brunoduc Can you please test this version with the PR from @tito:
> [script.module.inputstreamhelper-0.4.1-parsing-78a4a1d.zip](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3567014/script.module.inputstreamhelper-0.4.1-parsing-78a4a1d.zip)

Doesnt work for me, uninstalled Inputstreamhelper by deleting the directory,
Then installed zip, its downloading ChromeOS image without a problem,
But when extracting Widevine CDM it just doesnt go any further.
Any fix for this because I cant watch anything right now.

**Comment by mediaminister** [comment]
@Arathonk Please test the following package, it should work as it worked before 0.4.1
[script.module.inputstreamhelper-0.4.2-release-73fc720.zip](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3568850/script.module.inputstreamhelper-0.4.2-release-73fc720.zip)

Please report back if it worked.

**Comment by berturion** [comment]
Thanks @mediaminister. It works as expected. The progress bar remains motionless a while but at the end, it works.

**Comment by djskrat** [comment]
@mediaminister it is working after 0.4.2 manual update. Thank you

**Comment by brunoduc** [comment]
Work for me too.
manual update 0.4.1 to 0.4.2
[kodi.log](https://github.com/emilsvennesson/script.module.inputstreamhelper/files/3573419/kodi.log)
my wife thanks you !

**Comment by dagwieers** [comment]
So this is fixed with the release of v0.4.2.
